### PR TITLE
Remove uses of `_uniqueId` for React component keys

### PR DESCRIPTION
**Pull request by gjvoosten**
It would make the component always be recreated at every (re)render.
Replace this by a unique key, or in some cases an index (which is marginally better, provided the array being looped over is in some stable order).

### CODE REVIEW

File: client/src/components/AdvisorReports/AdvisorReportsRow.js [hunk 1]
```diff
@@ -1,21 +1,22 @@
 import Checkbox from "components/Checkbox"
-import _uniqueId from "lodash/uniqueId"
 import PropTypes from "prop-types"
 import React from "react"
 
 const _advisorStats = (columnGroups, statistics) => {
   const stats = []
-  columnGroups.forEach(group => {
+  columnGroups.forEach((group, index) => {
     let rowCell = statistics.filter(s => s.week === group)
```
Comment (cemalettin-simsoft) [inline client/src/components/AdvisorReports/AdvisorReportsRow.js:8]: `groups` are already unique values, why do we use index which can have side-effects later down the road? I suggest to use `key={`submitted_${group}`}` and remove `index`

File: client/src/components/AdvisorReports/AdvisorReportsTable.js [hunk 2]
```diff
@@ -47,11 +46,11 @@ const AdvisorReportsTable = ({ pageDispatchers, columnGroups, orgUuid }) => {
   }
 
   const advisors = data.advisorReportInsights
-  const rows = advisors.map(advisor => (
+  const rows = advisors.map((advisor, index) => (
     <AdvisorReportsRow
       row={advisor}
       columnGroups={columnGroups}
-      key={_uniqueId(`${advisor.uuid}_`)}
+      key={`${advisor.uuid}_${index}`}
```
Comment (cemalettin-simsoft) [inline client/src/components/AdvisorReports/AdvisorReportsTable.js:53]: `advisor.uuid` is already a unique value

File: client/src/components/AdvisorReports/AdvisorReportsRow.js [hunk 3]
```diff
@@ -1,21 +1,22 @@
...
+      stats.push(
+        <td key={`submitted_${index}`}>{rowCell.nrReportsSubmitted}</td>
+      )
+      stats.push(
+        <td key={`attended_${index}`}>{rowCell.nrEngagementsAttended}</td>
+      )
     } else {
-      stats.push(<td key={keySubmitted}>0</td>)
-      stats.push(<td key={keyAttended}>0</td>)
+      stats.push(<td key={`submitted_${index}`}>0</td>)
```
Comment (VassilIordanov) [inline client/src/components/AdvisorReports/AdvisorReportsRow.js:18]: shouldn't the key be a function of the group and not the index?

File: client/src/pages/dashboards/DiagramNode.js [hunk 4]
```diff
@@ -162,25 +161,27 @@ export const DiagramNodeWidget = ({ size, node, engine }) => {
...
+                  )
+                  questionConfig.showLegend = false
+                  return aggregationWidget ? (
+                    <AggregationWidgetContainer
+                      key={`assessment-${questionKey}`}
+                      fieldConfig={questionConfig}
+                      fieldName={questionKey}
+                      data={instantAssessmentResults}
+                      widget={aggregationWidget}
+                      widgetId={`${questionKey}-assessment-${index}`}
```
Comment (cemalettin-simsoft) [inline client/src/pages/dashboards/DiagramNode.js:180]: `questionKey` is already an unique value

**Review by cemalettin-simsoft — CHANGES_REQUESTED** [review]

File: client/src/components/AdvisorReports/AdvisorReportsTable.js (hunk 2 above)
Comment (gjvoosten) [reply client/src/components/AdvisorReports/AdvisorReportsTable.js:53]: The same advisor may appear more than once if they've moved organisations I believe.

Comment (cemalettin-simsoft) [reply client/src/components/AdvisorReports/AdvisorReportsTable.js:53]:
```
  const { done, result } = useBoilerplate({
    loading,
    error,
    modelName: "Organization",
    uuid: orgUuid,
    pageDispatchers
  })
  if (done) {
    return result
  }

  const advisors = data.advisorReportInsights
  const rows = advisors.map((advisor, index) => (
    <AdvisorReportsRow
      row={advisor}
      columnGroups={columnGroups}
      key={`${advisor.uuid}_${index}`}
    />
  ))
```
I thought we can't have duplicate advisors from single organization, or no ?

**Review by cemalettin-simsoft — APPROVED** [review]

**Pull request merged**